app/controllers/admin/mentors_controller.rb

```diff
@@ -1,4 +1,18 @@
 class Admin::MentorsController < ApplicationController
+
   def index
+    @user_ids = TrackMentorship.distinct.pluck(:user_id)
+    @users = User.where(id: @user_ids)
```

Comment:
This would be more efficient without the pluck on the line above. In the current version there are two SQL queries ran. If you replace `pluck` with `select` then the queries get merged and there's only one query. So you can change this to something like:

```User.where(id: TrackMentorship.distinct.select(:user_id))```

You can check in the rails server logs to see this in action.